articles/how-to-use-an-api-in-a-vuejs-project/index.md

```diff
@@ -0,0 +1,374 @@
+             console.log(error);
+         }
+
+       }
+   }
+}
+</script>
+```
+
+The above block of code in the `methods` property will be explained line by line. 
+
+```JavaScript
+async getData (){
+```
+
+A function named `getData()` is created . It is in this function the API will be called. The `async` keyword is prepended on the `getData` function to show that the function will make use of promise and we'll be using it to await to pause the execution of the function until the promise is resolved.
+
+```JavaScript
+try {
+    const response = await this.$http.get('http://jsonplaceholder.typicode.com/posts');
+```
+
+`try` property defines a block of code to be tested for errors as the code is executed. In the block of code `const response = await this.$http.get('http://jsonplaceholder.typicode.com/posts');`, a get request is made with the `get` keyword using axios i.e.`$http` to get data from the URL. 
+
+`await` is prepended to the request because the `get` function will return a promise. The data returned from the API after the promise is resolved and will be stored in the variable `response`.
+
+```JavaScript
+this.posts = response.data
+```
+
+The data we get from the request is saved to the 'posts' array which is created in the data property.
+
+```JavaScript
+catch (error) {
+    console.log(error);
+}
+```
+
+If any error occurs during the code execution, the error will be caught and logged in the console.
+
+After requesting data from the API, you will need to call it on a lifecycle hook. Here we will use the `created()` lifecycle hook, this is because we will be able to retrieve sensitive data and events that are active with the `created` hook.
+
+```JavaScript
+<template></template>
+
+<script>
+export default {
+data()  {
+posts: [ ]
+
+},
+
+methods: {
+async getData () {
+     try {
+              const response = await this.$http.get('http://jsonplaceholder.typicode.com/posts');
+             // JSON responses are automatically parsed.
+              this.posts = response.data
+              console.log(posts)
+           }
+          catch (error) {
+             console.log(error);
+         }
+
+       }
+   },
+
+
+created() {
+    this.getData();
+  },
+}
+</script>
+```
+
+We can now display the data in the template by looping through the posts using [v-for](https://vuejs.org/v2/guide/list.html) directive
+
+```JavaScript
+<template>
+     <div>
+         <div v-for="post in posts">
+             <h2>{{ post.title }}</h2>
+              <p>{{ post.body }}</p>
+         </div>
+      </div>
+</template>
+
+<script>
+export default {
+data()  {
+  posts: [ ]
+},
+
+methods: {
+async getData () {
+     try {
+              const response = await this.$http.get('http://jsonplaceholder.typicode.com/posts');
+              this.posts = response.data
+           }
+          catch (error) {
+             console.log(error);
+         }
+
+       }
+   },
+
+
+created() {
+    this.getData();
+  },
+}
+</script>
+```
```

Comment:
This is the error that I'm getting when I reach this point in the tutorial. Please fix them and make sure the application works before adding code snippets in the article. 

I'm getting similar errors on the previous steps too. Make sure the reader doesn't experience any error at any step in the article. 

![image](https://user-images.githubusercontent.com/45763204/120693773-11236380-c4c7-11eb-8c0f-c6721c030b32.png)